x-pack/legacy/plugins/maps/public/connected_components/layer_addpanel/source_editor/view.js

```diff
@@ -5,28 +5,22 @@
  */
 
 import React, { Fragment } from 'react';
-import { ALL_SOURCES } from '../../../layers/sources/all_sources';
+import { getLayerWizard } from '../../../layers/layer_wizard_registry';
 import { EuiSpacer, EuiPanel, EuiButtonEmpty } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 
 export const SourceEditor = ({
   clearSource,
-  sourceType,
+  layerWizardId,
   isIndexingTriggered,
   inspectorAdapters,
   previewLayer,
 }) => {
-  const editorProperties = {
-    onPreviewSource: previewLayer,
-    inspectorAdapters,
-  };
-  const Source = ALL_SOURCES.find(Source => {
-    return Source.type === sourceType;
-  });
-  if (!Source) {
-    throw new Error(`Unexpected source type: ${sourceType}`);
+  const layerWizard = getLayerWizard(layerWizardId);
```

Comment:
see other comment. Why not just pass in the actual wizard instance we got earlier through `getLayerWizards`